pkg/storageincentives/agent.go

```diff
@@ -254,35 +249,43 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 		}
 
 		// write the current phase only once
-
 		if currentPhase != prevPhase {
 
 			a.metrics.CurrentPhase.Set(float64(currentPhase))
 
 			a.logger.Info("entering phase", "phase", currentPhase.String(), "round", round, "block", block)
 
+			a.nodeState.SetPhase(currentPhase)
+			a.nodeState.SetRound(round)
+			a.nodeState.SetBlock(block)
+			a.nodeState.SaveStatus()
+
 			phaseEvents.Publish(currentPhase)
 			if currentPhase == claim {
 				phaseEvents.Publish(sample) // trigger sample along side the claim phase
 			}
 		}
 
 		prevPhase = currentPhase
-
 		mtx.Unlock()
+
 	}
 }
 
 func (a *Agent) reveal(ctx context.Context, storageRadius uint8, sample, obfuscationKey []byte) error {
 	a.metrics.RevealPhase.Inc()
 	err := a.contract.Reveal(ctx, storageRadius, sample, obfuscationKey)
+	a.nodeState.AddFee()
```

Comment:
I see that the `SaveStatus` is missing here, and calling it everywhere pollutes the code imo. 
maybe it's best that you simply save to disk on every call to the nodeState and remove all the `SaveStatus`

Reply:
It would be redundant to call `saveStatus` on every call because sometimes we are setting 2-3 values at once, and `saveStatus` will then be called 2-3 times as well.

Ideally, we should call `saveStatus` when we lose or win, or fail at some point.